.github/workflows/run_test_cases.yaml

```diff
@@ -10,26 +10,71 @@ on:
   pull_request:
 
 jobs:
-    run_proper_test:
+    prepare:
         runs-on: ubuntu-20.04
         container: emqx/build-env:erl23.3.4.9-3-ubuntu20.04
+        outputs:
+          ct_apps: ${{ steps.run_find_apps.outputs.ct_apps }}
+        steps:
+          - uses: actions/checkout@v2
+            with:
+              path: source
+              fetch-depth: 0
+          - name: find_ct_apps
+            working-directory: source
+            id: run_find_apps
+            # emqx_plugin_libs doesn't have a test suite -> excluded from app list
+            run: |
+              ct_apps="$(./scripts/find-apps.sh --json | jq -c 'del (.[] | select (. == "apps/emqx_plugin_libs"))')"
+              echo "ct-apps: $ct_apps"
+              echo "::set-output name=ct_apps::$ct_apps"
```

Comment:
I know we still use this syntax everywhere, but github is already issuing warnings https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/